source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -893,18 +893,22 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         IRBuilder builder(inst);
         builder.setInsertBefore(inst);
         auto funcType = as<IRFuncType>(funcValue->getFullType());
+        bool argsChanged = false;
         for (UInt i = 0; i < inst->getArgCount(); i++)
         {
             auto arg = inst->getArg(i);
             auto paramType = funcType->getParamType(i);
-            if (as<IRPtrType>(paramType))
+            if (auto ptrType = as<IRPtrType>(paramType))
             {
-                // If the parameter has an explicit pointer type,
-                // then we know the user is using the variable pointer
-                // capability to pass a true pointer.
-                // In this case we should not rewrite the call.
-                newArgs.add(arg);
-                continue;
+                if (ptrType->getAddressSpace() == AddressSpace::UserPointer)
+                {
+                    // If the parameter has an explicit pointer type,
+                    // then we know the user is using the variable pointer
+                    // capability to pass a true pointer.
+                    // In this case we should not rewrite the call.
+                    newArgs.add(arg);
+                    continue;
+                }
```

Comment:
This is going to overlap some of what I'm working on, but won't be too hard to handle in a merge.

The only thing that I think could be done to make things easier might be to define a utility function in `slang-ir-insts.h` along the lines of :

    IRPtrType* asExplicitPtrType();

That would make it easier to catch cases that are currently using the `AddressSpace::UserPointer` thing to check for explicit pointers from the user's program.